### ISSUE TITLE
Latest Fable compiler v4.14.0 breaks the getTodos API call in the latest SAFE stack template v5.04 - when using latest Fable.Remoting v7.31

### ISSUE THREAD

**Issue by PiotrW3106** [fable-compiler/Fable#3784]
### Description

Hi
I think I have found a bug in either the Fable compiler and or Fable.Remoting but I am stuck as to what to do next - where to file the bug report i.e. against the Fable compiler or Fable.Remoting as I cannot figure out exactly where the bug is and I am now stuck.

### Repro code
To reproduce the bug take the latest SAFE stack template (SAFE-Stack Web App v5.0.4 SAFE F# Web/Saturn/Azure/Fable/Elmish )

Restore the tools then dotnet run it (without any modification) then the whole application works correctly.

However the SAFE Stack is using an old version of the Fable compiler (v 4.9.0) and also uses an slightly older version of Fable.Remoting v7.30 

If I upgrade (using paket) to the latest Fable.Remoting v7.31 and then the latest Fable compiler v4.14.0 then a problem occurs, namely the getTodos in the ITodosApi stops working when called from the Init function in Index.fs line 22, (silently – no error messages – it just does not make the call to the server when I examine the networking in the browser) i.e. it fails to get the initial hard-coded list of todos from the server.

Note: the  addTodo: Todo -> Async<Todo>  call works correctly and I can add new todos.

Trying to narrow down the root cause it seems that using the Fable compiler v4.12.2 works correctly with Fable.Remoting v7.30 and v7.31 but upgrading the compiler to either v4.13.0 or v4.14.0 causes the bug to appear. 

Downgrading the compiler back to v4.12.2 without changing anything else makes it work again (i.e. leaving Fable.Remoting at v7.31). 

Thus I think it is more probable a Fable compiler issue? 

Interestingly if I change the function signature from getTodos: unit -> Async<Todo list> to getTodos: string -> Async<Todo list> then fix up the code by giving it a random string rather than unit in the Init function causes it to work correctly again with the latest Fable compiler v4.14.0 and Fable.Remoting v7.31. 

Apologises I have not been able to debug this further. 

thanks

### Repro code
Sorry I have not been able to provide code fragments. 

Please provide the F# code to reproduce the problem or a link to the [REPL](https://fable.io/repl/).
Ideally, it should be possible to easily turn this code into a unit test.

### Expected and actual results

Please provide the expected and actual results.

### Related information

* Fable version: `dotnet fable --version`
* Operating system - Ubuntu 20.04.6 LTS using latest .net  8.0.202 sdk

**Comment by cr3wdayt5p** [comment]
Seems to be a bug in `Fable.Remoting.Client`:
https://github.com/Zaid-Ajaj/Fable.Remoting/issues/361

**Comment by MangelMaxime** [comment]
Indeed, this issue is caused by the changes we made to align to .NET behaviour when accessing an array out of bound.

Before, it was returning `undefined` and now it will throw an exception.

I am closing this issue as the fix is to have a new release of Fable.Remoting, but if the bug persist after please feel free to re-open.
